pkgs/build-support/gradle/update-deps.nix

```
@@ -0,0 +1,154 @@
+{ lib
+, pkgs
+, stdenv
+, writeScript
+, writeShellScript
+, nix
+, gradle
+, openssl
+, mitm-cache
+, jdk
+, nsjail
+, path
+, python3
+}:
+
+# path to the derivation attr (usually the package name)
+{ attrPath ? pname
+# pname (attrPath synonym)
+, pname ? ""
+# the Nix code to get the derivation (this will be put in Bash double quotes)
+# this is relevant for downstream (non-nixpkgs) users
+, code ? "with import ./. { }; ($attrPath)"
+# unlike gradleFlags in normal packages, these flags are escaped so they are allowed to contain spaces
+# also, gradleFlags from configurePhase are used by default
+, gradleFlags ? []
+# attribute overrides for the "source" derivation
+, attrOverrides ? {}
+# run before fetching the dependencies (in sandbox)
+, preBuild ? ""
+# run before starting the sandbox for fetching dependencies
+, preBuildNoSandbox ? ""
+# the gradle commands for fetching the necessary dependencies
+, gradleCommands ? "gradle nixDownloadDeps"
+# run after fetching the dependencies (in sandbox)
+, postBuild ? ""
+# run after the sandbox for fetching dependencies has stopped
+, postBuildNoSandbox ? ""
+, mitmCachePort ? 1337
+# deps path (relative to the package directory), or absolute
+, depsPath ? "deps.json"
+# extra packages to make available in the sandbox
+, availablePackages ? [ ]
+}:
+
+let
+  keep = [ "MITM_CACHE_PORT" "MITM_CACHE_KEYSTORE" "MITM_CACHE_KS_PWD" "MITM_CACHE_CA" ];
+  availablePackages' = map (drv: ''(builtins.storePath "${drv}")'') availablePackages;
+  gradleScript = writeScript "gradle-commands.sh" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell --pure ${lib.concatMapStringsSep " " (x: "--keep ${x}") keep} -i bash -p coreutils ${lib.concatStringsSep " " availablePackages'}
+    set -eo pipefail
+    source="$1"
+    gradleArgs=(
+      ${lib.escapeShellArgs gradleFlags}
+      --no-daemon
+      --console plain
+      --init-script ${./init-deps.gradle}
+      -Dhttp.proxyHost="127.0.0.1"
+      -Dhttps.proxyHost="127.0.0.1"
+      -Dhttp.proxyPort="$MITM_CACHE_PORT"
+      -Dhttps.proxyPort="$MITM_CACHE_PORT"
+      -Djavax.net.ssl.trustStore="$MITM_CACHE_KEYSTORE"
+      -Djavax.net.ssl.trustStorePassword="$MITM_CACHE_KS_PWD"
+    )
+    unset MITM_CACHE_KEYSTORE
+    unset MITM_CACHE_KS_PWD
+    function gradle() {
+      ${gradle}/bin/gradle $gradleFlags "''${gradleFlagsArray[@]}" "''${gradleArgs[@]}" "$@"
+    }
+    export GRADLE_USER_HOME="$(mktemp -d)"
+    export TERM=dumb
+    trap "rm -rf '$GRADLE_USER_HOME'" EXIT
```

```suggestion
    trap "rm -rf '$GRADLE_USER_HOME'" SIGINT SIGTERM ERR EXIT
```

in case of errors we don't want to fill up the tmp directory